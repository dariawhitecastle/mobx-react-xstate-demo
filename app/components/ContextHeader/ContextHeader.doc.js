import { describe, PropTypes } from 'react-desc'
import { intlShape } from 'react-intl'

export const doc = ContextHeader => {
  const DocumentedContextHeader = describe(ContextHeader)
    .description('Subheader for the context of each page, e.g. Profile, Dashboard, etc')
    .usage(
      `import { ContextHeader } from 'components/ContextHeader';
      <ContextHeader headerMain={messages.header} name="Touring company" status="Normal"/>`,
    )

  DocumentedContextHeader.propTypes = {
    name: PropTypes.string.isRequired.description('Name of the organization').defaultValue(''),
    status: PropTypes.string.isRequired
      .description('Status of current operations')
      .defaultValue(''),
    headerMain: PropTypes.shape({
      id: PropTypes.string.isRequired,
      defaultMessage: PropTypes.string.isRequired,
    })
      .description('Id and title for the header')
      .defaultValue({
        id: 'header',
        defaultMessage: 'This is the ContextHeader component!',
      }),
    intl: intlShape.isRequired,
  }

  return DocumentedContextHeader
}
