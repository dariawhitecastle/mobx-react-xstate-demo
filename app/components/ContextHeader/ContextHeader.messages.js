/*
 * ContextHeader Messages
 *
 * This contains all the text for the ContextHeader component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.ContextHeader'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ContextHeader component!',
  },
  currentOperations: {
    id: `${scope}.currentOperations`,
    defaultMessage: 'Current Operations:',
  },
  changeStatus: {
    id: `${scope}.changeStatus`,
    defaultMessage: 'Change Status',
  },
})
