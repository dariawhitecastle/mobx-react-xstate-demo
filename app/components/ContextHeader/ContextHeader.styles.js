import styled from 'styled-components'
import { Button } from '../Button'

export const StyledPlainButton = styled(Button)`
  background-color: transparent;
`
