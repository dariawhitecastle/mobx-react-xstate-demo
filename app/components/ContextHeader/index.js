export { default as ContextHeader } from './ContextHeader'
