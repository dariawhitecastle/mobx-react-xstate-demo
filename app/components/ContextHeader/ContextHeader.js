/* eslint-disable react/prop-types */
/**
 * ContextHeader
 */

import React, { useContext } from 'react'
import { injectIntl } from 'react-intl'

import { Box } from '../Box'
import { Header } from '../Header'
import { Text } from '../Text'

import { StyledPlainButton } from './ContextHeader.styles'
import messages from './ContextHeader.messages'
import colors from '../../utils/colors'
import { fontSizes } from '../../utils/sizes'

import { UserStoreContext } from '../../stores/UserStore'

const ContextHeader = ({ headerMain, intl, name, status, premiumProp = true }) => {
  /** @todo remove hardcoded premium value here */
  const { premium } = { premium: premiumProp, ...useContext(UserStoreContext) }
  const { currentOperations, changeStatus } = messages
  return (
    <Box direction="row" fill="horizontal" justify="between">
      <Header level="3" color={colors.secondary} margin={{ bottom: 'xsmall' }}>
        {intl.formatMessage(headerMain)}
      </Header>
      <Box>
        <Text size="large" alignSelf="end" color={colors.secondary}>
          {name}
        </Text>
        {premium && (
          <Box direction="row">
            <Text alignSelf="center" color={colors.primaryText} size={fontSizes.small}>
              {intl.formatMessage(currentOperations)}
            </Text>
            <StyledPlainButton plain margin={{ left: '6px' }}>
              <Text color={colors.secondary} size={fontSizes.small}>
                {status.toUpperCase()}
              </Text>
            </StyledPlainButton>
            <StyledPlainButton plain margin={{ left: '16px' }}>
              <Text color={colors.primary} size={fontSizes.xsmall}>
                {intl.formatMessage(changeStatus)}
              </Text>
            </StyledPlainButton>
          </Box>
        )}
      </Box>
    </Box>
  )
}

ContextHeader.propTypes = {}

export default injectIntl(ContextHeader)
