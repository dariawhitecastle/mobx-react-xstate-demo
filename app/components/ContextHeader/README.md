<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## InjectIntl(ContextHeader)
Subheader for the context of each page, e.g. Profile, Dashboard, etc

## Usage

```javascript
import { ContextHeader } from 'components/ContextHeader';
      <ContextHeader headerMain={messages.header} name="Touring company" status="Normal"/>
```

## Properties

**name**

Required. Name of the organization

```
string
```

**status**

Required. Status of current operations

```
string
```

**headerMain**

Id and title for the header Defaults to `{
  "id": "header",
  "defaultMessage": "This is the ContextHeader component!"
}`.

```
{
  id: string,
  defaultMessage: string
}
```
  