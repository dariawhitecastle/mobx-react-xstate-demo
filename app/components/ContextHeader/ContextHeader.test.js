/**
 *
 * Tests for ContextHeader
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import messages from './ContextHeader.messages'
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ContextHeader from './ContextHeader'
import { DEFAULT_LOCALE } from '../../i18n'

const renderComponent = (props = {}) =>
  render(
    <IntlProvider locale="en">
      <ContextHeader headerMain={messages.header} {...props} />
    </IntlProvider>,
  )

test('it renders and matches snapshot', () => {
  const { container } = renderComponent({ name: 'Touring company', status: 'Normal' })
  expect(container).toMatchSnapshot()
})

test('It does not log to the console', () => {
  const spy = jest.spyOn(global.console, 'log')
  render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <ContextHeader headerMain={messages.header} name="Touring company" status="Normal" />
    </IntlProvider>,
  )
  expect(spy).not.toHaveBeenCalled()
})
