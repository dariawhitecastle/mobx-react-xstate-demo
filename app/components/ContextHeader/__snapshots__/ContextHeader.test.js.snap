// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 laoUSJ"
  >
    <h3
      class="StyledHeading-sc-1rdh4aw-0 eoFZBW"
    >
      This is the ContextHeader component!
    </h3>
    <div
      class="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <span
        class="StyledText-sc-1sadyjn-0 eIjKJy"
      >
        Touring company
      </span>
      <div
        class="StyledBox-sc-13pk1d4-0 ffeAql"
      >
        <span
          class="StyledText-sc-1sadyjn-0 hfhDKB"
        >
          Current Operations:
        </span>
        <button
          class="ContextHeaderstyles__StyledPlainButton-v85b05-0 frwaHS StyledButton-sc-323bzc-0 ggkrQe"
          type="button"
        >
          <span
            class="StyledText-sc-1sadyjn-0 blCtzG"
          >
            NORMAL
          </span>
        </button>
        <button
          class="ContextHeaderstyles__StyledPlainButton-v85b05-0 frwaHS StyledButton-sc-323bzc-0 PFIEG"
          type="button"
        >
          <span
            class="StyledText-sc-1sadyjn-0 dFayqK"
          >
            Change Status
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
